app/main.py

```diff
@@ -1 +1,51 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self._health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def __str__(self) -> str:
+        return (
+            "{"
+            f"Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}"
+            "}"
+        )
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    @property
+    def health(self) -> int:
+        return self._health
+
+    @health.setter
+    def health(self, new_health: int) -> None:
+        if new_health > 0:
+            self._health = new_health
+        else:
+            self._health = 0
+            Animal.alive.remove(self)
```

Comment:
Before removing `self` from `Animal.alive`, you should check if `self` is in the list to avoid potential errors. Consider using `if self in Animal.alive:` before attempting to remove it.